sqlrunner/main.py

```diff
@@ -24,9 +24,13 @@ def main(args):
     if not are_t1oos_handled(sql_query):
         raise RuntimeError("T1OOs are not handled correctly")
 
-    if args["dsn"] is None and args["dummy_data_file"] is not None:
-        # Bypass the database
-        results = args["dummy_data_file"]
+    if args["dsn"] is None:
+        if args["dummy_data_file"] is None:
+            # Output dummy data file with header but no rows
+            results = generate_column_headers(sql_query)
```

Comment:
I think the comment is a bit whiffy. What about...

```suggestion
            results = iter([{x: None for x in generate_column_headers(sql_query)}])
```

...after reworking `generate_column_headers`?